modules/tinymce/src/plugins/table/test/ts/browser/ui/TableDialogStyleWithCssTest.ts

```diff
@@ -0,0 +1,200 @@
+      });
+
+      it('sets either td "padding" style or cellpadding attribute, based on integrator preference', async () => {
+        const editor = hook.editor();
+        initializeTable(editor, {}, {}, {});
+
+        await TableTestUtils.pOpenTableDialog(editor);
+
+        TableTestUtils.setDialogValues({
+          cellspacing: 0,
+          cellpadding: '30px'
+        }, false, generalSelectors);
+
+        await TableTestUtils.pClickDialogButton(editor, true);
+
+        if (spec.style_by_css) {
+          assertTable(editor, { cellpadding: '' }, {}, { padding: '30px' });
+        } else {
+          assertTable(editor, { cellpadding: '30px' }, {}, { padding: '' });
+        }
+      });
+
+      it('falls back to cellspacing attribute if no CSS is defined', async () => {
+        const editor = hook.editor();
+        initializeTable(editor, { cellspacing: '5px' }, {}, {});
+
+        await TableTestUtils.pOpenTableDialog(editor);
+
+        TableTestUtils.assertDialogValues({
+          cellspacing: '5px',
+          cellpadding: 0
+        }, false, generalSelectors);
+
+        await TableTestUtils.pClickDialogButton(editor, false);
+      });
+
+      it('falls back to table border-spacing style if no cellspacing attribute is defined', async () => {
+        const editor = hook.editor();
+        initializeTable(editor, {}, { 'border-spacing': '5px' }, {});
+
+        await TableTestUtils.pOpenTableDialog(editor);
+
+        TableTestUtils.assertDialogValues({
+          cellspacing: '5px',
+          cellpadding: 0
+        }, false, generalSelectors);
+
+        await TableTestUtils.pClickDialogButton(editor, false);
+      });
+
+      it('uses the integrators preference when both table "border-spacing" style and cellspacing attribute are defined', async () => {
+        const editor = hook.editor();
+        initializeTable(editor, { cellspacing: '5px' }, { 'border-spacing': '10px' }, {});
+
+        await TableTestUtils.pOpenTableDialog(editor);
+
+        TableTestUtils.assertDialogValues({
+          cellspacing: spec.style_by_css ? '10px' : '5px',
+          cellpadding: 0
+        }, false, generalSelectors);
+
+        await TableTestUtils.pClickDialogButton(editor, false);
+      });
+
+      it('sets either table "border-spacing" style or cellspacing attribute, based on integrator preference', async () => {
+        const editor = hook.editor();
+        initializeTable(editor, {}, {}, {});
+
+        await TableTestUtils.pOpenTableDialog(editor);
+
+        TableTestUtils.setDialogValues({
+          cellspacing: '5px',
+          cellpadding: 0
+        }, false, generalSelectors);
+
+        await TableTestUtils.pClickDialogButton(editor, true);
+
+        if (spec.style_by_css) {
+          assertTable(editor, { cellspacing: '' }, { 'border-spacing': '5px' }, {});
+        } else {
+          assertTable(editor, { cellspacing: '5px' }, { 'border-spacing': '' }, {});
```

Comment:
Apparently this sort of thing gets normalised to `5` on IE. How do we normally handle things like that in our tests?

Reply:
We set a valid value in the first place? `5px` is not a valid value for the cellspacing attribute on line 186 when using `table_style_by_css: false` as it needs to be a number not a CSS dimension.